### ISSUE TITLE
ELI5: ExecuteArgument

### ISSUE THREAD

**Issue by keekuu** [oliverschwendener/ueli#112]
Hi there, I'm new but I've reviewed the Issues related to custom command threads. I'm sorry I know this might be obvious but it is not to me and it wasn't explicitly explained anywhere. 

I want my own custom search style. I dont want to type the question mark. Instead of google searching by typing in "g? SEARCHTERM" I just want to be able to do "g SEARCHTERM". 

I think I can do this with custom commands which I know is in the settings but what isnt clear to me is how to do the "ExecuteArgument" part.  I've googled "ExecuteArgument" hoping to give me more context as to the language it needs to be written in or some common examples but did not find anything. Could you please explain this? What language is an execute argument? Is there an external reference link for common types of commands?

**Comment by oliverschwendener** [comment]
> I want my own custom search style. I dont want to type the question mark. Instead of google searching by typing in "g? SEARCHTERM" I just want to be able to do "g SEARCHTERM".

This is currently not possible. In the next major update you should be able to create less restricted prefixes.

> I think I can do this with custom commands which I know is in the settings but what isnt clear to me is how to do the "ExecuteArgument" part. I've googled "ExecuteArgument" hoping to give me more context as to the language it needs to be written in or some common examples but did not find anything. Could you please explain this? What language is an execute argument? Is there an external reference link for common types of commands?

The "execution argument" is the information for ueli "what to execute". For example if you want to open a text file with Visual Studio Code on Windows the execution argument would be:

```
code C:\Users\Oliver\project\my-file.txt
```

Hope this is solves your problem.

**Comment by keekuu** [comment]
Thanks Oliver. Looking forward to the update. 

I understood the concept of "execute argument" but I struggled with understanding how to compose one, what syntax/language to use.  I think I understand now. I needed to learn about [Run commands](https://en.wikipedia.org/wiki/Run_command) and [URIs](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier).
